### PR TITLE
Use actor-of-interest in DistToDestination metric and reward function

### PR DESCRIPTION
**Pull request by Adaickalavan**
- Reward function in platoon RL example retrieves actor-of-interest from marked neighborhood vehicles.
- `dist_to_destination` metric cost function computes the route distance at the end of an episode using the vehicle-of-interest's last position.

### CODE REVIEW

File: smarts/env/gymnasium/wrappers/metric/costs.py [hunk 1]
```diff
@@ -484,7 +494,7 @@ class CostFuncsBase:
     # fmt: off
     collisions: Callable[[], Callable[[RoadMap, VehicleIndex, Done, Observation], Costs]] = _collisions
     comfort: Callable[[], Callable[[RoadMap, VehicleIndex, Done, Observation], Costs]] = _comfort
-    dist_to_destination: Callable[[Point,float], Callable[[RoadMap, VehicleIndex, Done, Observation], Costs]] = _dist_to_destination
+    dist_to_destination: Callable[[str, float, Point], Callable[[RoadMap, VehicleIndex, Done, Observation], Costs]] = _dist_to_destination
```
Comment (Gamenot) [inline smarts/env/gymnasium/wrappers/metric/costs.py:497]: Do we have a description of what these metrics are intended to do (even if the implementation is private?)

**Review by Gamenot — APPROVED** [review]

File: examples/rl/platoon/train/reward.py [hunk 2]
```diff
@@ -59,13 +59,10 @@ def step(self, action):
...
         reward = {agent_id: np.float64(0) for agent_id in obs.keys()}
 
-        leader_name = "Leader-007"
         leader = None
         max_agent_steps_completed = 0
         for agent_id, agent_obs in obs.items():
-            neighbor_vehicles = _get_neighbor_vehicles(
-                obs=agent_obs, neighbor_name=leader_name
-            )
+            neighbor_vehicles = _get_leader(obs=agent_obs)
```
Comment (Gamenot) [inline examples/rl/platoon/train/reward.py:65]: I think `_get_leader()` is a misleading name for the method. It should indicate plurality and also what it is getting. I also think that `neighbor_vehicles` is probably not specific enough.

```suggestion
            interest_vehicles = _get_interest_vehicles(obs=agent_obs)
```

File: smarts/env/gymnasium/wrappers/metric/costs.py (hunk 1 above)
Comment (Adaickalavan) [reply smarts/env/gymnasium/wrappers/metric/costs.py:497]: Although not available currently, a description of the metrics is on the list of things to do next. It is tracked in issue #1741.

File: examples/rl/platoon/train/reward.py (hunk 2 above)
Comment (Adaickalavan) [reply examples/rl/platoon/train/reward.py]: I have changed it accordingly.

File: smarts/env/gymnasium/wrappers/metric/costs.py [hunk 3]
```diff
@@ -352,7 +362,7 @@ def func(
         step = step + 1
 
         if not done:
-            return Costs(steps=-1)
+            return Costs(steps=-1e8)
```
[the remark this answers is not in the document]
Comment (Adaickalavan) [reply smarts/env/gymnasium/wrappers/metric/costs.py]: A default value was needed to be returned. This default value would not affect the metrics as it is not used. In order to make the error obvious in the unlikely event of the default value somehow being used in the code, a large negative value, which is outside the valid set of numbers, was used as the default. The default values are now updated to `-np.inf`.

**Review by saulfield — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -61,6 +61,8 @@ Copy and pasting the git commit messages is __NOT__ enough.
 - Driving SMARTS 2023.3 benchmark and the metrics module now uses `actor_of_interest_re_filter` from scenario metadata to identify the lead vehicle.
 - Included `RelativeTargetPose` action space to the set of allowed action spaces in `platoon-v0` env.
 - `Collision.collidee_id` now gives the vehicle id rather than the name of the owner of the vehicle (usually the agent id.) `Collision.collidee_owner_id` now provides the id of the controlling `agent` (or other controlling entity in the future.) This is because 1) `collidee_id` should refer to the body and 2) in most cases the owner name would be `None`.
+- Reward function in platoon RL example retrieves actor-of-interest from marked neighborhood vehicles.
+- `dist_to_destination` metric cost function computes the route distance at the end of an episode using the vehicle-of-interest's last position.
```
[the remark this answers is not in the document]
Comment (Adaickalavan) [reply CHANGELOG.md]: I have pushed new changes which should account for early termination by pre-computing the total route distance upon episode reset for vehicle of interest from social-agents, SMARTS traffic provider, SUMO traffic provider, and history traffic provider.

**Pull request merged**